src/services/fs/nfs.rs

```diff
@@ -0,0 +1,740 @@
+/// # std::io::Result::Ok(()) });
+/// ```
+pub async fn remove_dir<P: AsRef<Path>>(path: P) -> io::Result<()> {
+    let path = path.as_ref().to_owned();
+    spawn_blocking(move || std::fs::remove_dir(&path)).await
+}
+
+/// Removes a file.
+///
+/// # Errors
+///
+/// An error will be returned in the following situations:
+///
+/// * `path` does not point to an existing file.
+/// * The current process lacks permissions to remove the file.
+/// * Some other I/O error occurred.
+///
+/// # Examples
+///
+/// ```no_run
+/// # futures::executor::block_on(async {
+/// opendal::services::fs::nfs::remove_file("a.txt").await?;
+/// # std::io::Result::Ok(()) });
+/// ```
+pub async fn remove_file<P: AsRef<Path>>(path: P) -> io::Result<()> {
+    let path = path.as_ref().to_owned();
+    spawn_blocking(move || std::fs::remove_file(&path)).await
+}
+
+/// An open file on the filesystem.
+///
+/// Depending on what options the file was opened with, this type can be used for reading and/or
+/// writing.
+///
+/// Files are automatically closed when they get dropped and any errors detected on closing are
+/// ignored. Use the [`sync_all()`][`File::sync_all()`] method before dropping a file if such
+/// errors need to be handled.
+///
+/// **NOTE:** If writing to a file, make sure to call
+/// [`flush()`][`futures::io::AsyncWriteExt::flush()`], [`sync_data()`][`File::sync_data()`],
+/// or [`sync_all()`][`File::sync_all()`] before dropping the file or else some written data
+/// might get lost!
+///
+/// # Examples
+///
+/// Create a new file and write some bytes to it:
+///
+/// ```no_run
+/// use opendal::services::fs::nfs::File;
+/// use futures::io::AsyncWriteExt;
+///
+/// # futures::executor::block_on(async {
+/// let mut file = File::create("a.txt").await?;
+///
+/// file.write_all(b"Hello, world!").await?;
+/// file.flush().await?;
+/// # std::io::Result::Ok(()) });
+/// ```
+///
+/// Read the contents of a file into a vector of bytes:
+///
+/// ```no_run
+/// use opendal::services::fs::nfs::File;
+/// use futures::io::AsyncReadExt;
+///
+/// # futures::executor::block_on(async {
+/// let mut file = File::open("a.txt").await?;
+///
+/// let mut contents = Vec::new();
+/// file.read_to_end(&mut contents).await?;
+/// # std::io::Result::Ok(()) });
+/// ```
+pub struct File {
+    /// Always accessible reference to the file.
+    file: Arc<std::fs::File>,
+
+    /// Performs blocking I/O operations on a thread pool.
+    unblock: Mutex<Unblock<ArcFile>>,
+
+    /// Logical file cursor, tracked when reading from the file.
+    ///
+    /// This will be set to an error if the file is not seekable.
+    read_pos: Option<io::Result<u64>>,
+
+    /// Set to `true` if the file needs flushing.
+    is_dirty: bool,
+}
+
+impl File {
+    /// Creates an async file from a blocking file.
+    fn new(inner: std::fs::File, is_dirty: bool) -> File {
+        let file = Arc::new(inner);
+        let unblock = Mutex::new(Unblock::new(ArcFile(file.clone())));
```

Comment:
Unblock has been proven to be buggy. I don't like the idea of introducing `Unblock` back.

Can we use `nuclei::Handle::<File>` directly?

Reply:
> Can we use nuclei::Handle::<File> directly?

Yes

```rust
        let fo = OpenOptions::new()
            .read(true)
            .write(true)
            .open(&path)
            .unwrap();
        let mut file = Handle::<File>::new(fo).unwrap();
```

but , `test_reader` will overtime

Reply:
only iouring feature is buggy, we can only use epoll on linux.